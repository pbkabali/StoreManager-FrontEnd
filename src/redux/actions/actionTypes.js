export const _SUCCEEDED = "_SUCCEEDED";
export const _FAILED = "_FAILED";
export const _STARTED = "_STARTED";
export const _ENDED = "ENDED";
export const LOGIN_ACTION = "LOGIN_ACTION";
