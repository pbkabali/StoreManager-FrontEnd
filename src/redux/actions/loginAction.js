import * as actions from "./actionTypes";
import axios from "axios";
import { createActionThunk } from "redux-thunk-actions";

const login = payload => {
  return axios.post(
    "https://polos-store-manager.herokuapp.com/api/v1/auth/login",
    payload
  );
};

export const loginAction = createActionThunk(actions.LOGIN_ACTION, login, true);
