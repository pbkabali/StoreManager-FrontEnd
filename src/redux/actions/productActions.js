import * as actions from "./actionTypes";
import axios from "axios";
import { createActionThunk } from "redux-thunk-actions";

const getProducts = () => {
  return axios.get(
    "https://polos-store-manager.herokuapp.com/api/v1/products",
    { headers: { Authorization: "Bearer " + localStorage.getItem("token") } }
  );
};

export const getProductsAction = createActionThunk(
  actions.FETCH_PRODUCTS,
  getProducts,
  true
);
