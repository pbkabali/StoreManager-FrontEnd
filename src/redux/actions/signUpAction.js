import * as actions from "./actionTypes";
import axios from "axios";
import { createActionThunk } from "redux-thunk-actions";

const signUp = payload => {
  return axios.post(
    "https://polos-store-manager.herokuapp.com/api/v1/auth/signup",
    payload
  );
};

export const signUpAction = createActionThunk(
  actions.SIGNUP_ACTION,
  signUp,
  true
);
