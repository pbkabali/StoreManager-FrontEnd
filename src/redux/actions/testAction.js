import * as actions from './actionTypes';

export const testAction = () => {
    return {
      type: actions.TEST_ACTION
    };
  };
