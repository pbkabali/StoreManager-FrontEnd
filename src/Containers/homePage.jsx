import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import {
  Grid,
  Button,
  Form,
  GridColumn,
  Segment,
  Header
} from "semantic-ui-react";
import { connect } from "react-redux";
import { loginAction } from "../redux/actions/loginAction";

export class HomePage extends Component {
  state = {
    username: "user1",
    password: "user123"
  };

  onInputChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  loginUser = () => {
    const { username, password } = this.state;
    const payload = {
      username,
      password
    };
    this.props.loginAction(payload);
    this.setState({
      redirect: true
    });
  };

  renderRedirect = () => {
    if (this.props.loggedIn) {
      localStorage.setItem("token", this.props.loginResponse.access_token);
      return <Redirect to="/products" />;
    }
  };
  render() {
    return (
      <div
        style={{
          marginTop: "180px",
          marginBottom: "140px"
        }}
      >
        <Grid>
          <Grid.Row>
            <GridColumn width={5} />
            <GridColumn width={6} textAlign="center">
              <Segment color="red" padded="very">
                {this.renderRedirect()}
                <Header color="grey" as="h2" textAlign="center">
                  Welcome to Polos Store Manager App!
                </Header>
                <br />
                <Header color="grey" as="h4" textAlign="center">
                  Enter your login credentials
                </Header>
                <br />
                <Form width={2}>
                  <Form.Input
                    icon="user"
                    iconPosition="left"
                    placeholder="Username"
                    required
                    onChange={this.onInputChange}
                    name="username"
                  />
                  <br />
                  <Form.Input
                    type="password"
                    icon="lock"
                    iconPosition="left"
                    placeholder="Password"
                    required
                    onChange={this.onInputChange}
                    name="password"
                  />
                </Form>
                <br />
                <br />
                <Button
                  size="large"
                  onClick={this.loginUser}
                  loading={this.props.loading}
                >
                  Login
                </Button>
              </Segment>
            </GridColumn>
            <GridColumn width={5} />
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    loading: state.login.loading,
    loginResponse: state.login.loginResponse,
    loggedIn: state.login.isLoggedIn
  };
};

const mapDispatchToProps = dispatch => {
  return {
    loginAction: payload => dispatch(loginAction(payload))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomePage);
