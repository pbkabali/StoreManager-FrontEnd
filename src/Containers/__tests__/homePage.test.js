import React from "react";
import { HomePage } from "../homePage";
import { shallow } from "enzyme";
import { loginAction } from "../../redux/actions/loginAction";

describe("login container tests", () => {
  const onSubmitMock = jest.fn();
  const props = {
    loginUser: onSubmitMock,
    loginAction: onSubmitMock
  };
  const wrapper = shallow(<HomePage {...props} />);
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls submit function", () => {
    wrapper.find("Button").simulate("click");
    expect(onSubmitMock).toHaveBeenCalled();
  });
});
