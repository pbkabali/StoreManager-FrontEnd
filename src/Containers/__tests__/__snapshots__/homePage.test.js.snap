// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login container tests renders correctly 1`] = `ShallowWrapper {}`;
